service/controller/clusterapi/v17/resources/machinedeploymentstatus/delete.go

```diff
@@ -2,8 +2,103 @@ package machinedeploymentstatus
 
 import (
 	"context"
+
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/controllercontext"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
```

Comment:
Is there any difference between EnsureCreated/EnsureDeleted? If no they why EnsureDeleted doesn't call EsnureCreated or a common private method?

Reply:
It is identical but redirecting one within the other makes the logs and error stacks weird. I am fine with copying.

Reply:
I'd strongly prefer to have extracted common private method. This is confusing.

Reply:
I am not sure how useful that would be because we do not want to have control flow primitives outside of the interface methods of the resource. So it would be very questionable what we put in such a common method. When you feel like it, please provide a PR and suggest something. I would like to not pursue that. Alternatively we can pair and talk it through.